src/data/polynomial/denominators.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.erase_lead
+/-!
+# Denominators of evaluation of polynomials at ratios
+
+Let `i : R → K` be a homomorphism of semirings.  Assume that `K` is commutative.  If `a` and
+`b` are elements of `R` such that `i b ∈ K` is invertible, then for any polynomial
+`f ∈ polynomial R` the "mathematical" expression `b ^ f.nat_degree * f (a / b) ∈ K` is in
+the image of the homomorphism `i`.
+-/
+
+open polynomial finset
+
+section clearing_denominators
+
+variables {R K : Type*} [semiring R] [comm_semiring K] {i : R →+* K}
+variables {a b : R} {bi : K}
+-- TODO: use hypothesis (ub : is_unit (i b)) to work with localizations.
+
+lemma is_integer_eval_zero (N : ℕ) (a b : R) :
+  ∃ D : R, i D = i b ^ N * eval (i a * bi) (polynomial.map i 0) :=
```

Comment:
I'm wondering, should we give a name to the expression
`i b ^ N * eval (i a * bi) (polynomial.map i foobar)`?
I don't have a good suggestion of the top of my head...

Reply:
Something with numerator?  I think that the more important one is the one with `f.nat_degree` instead of `N`, hence the numerator suggestion...

Reply:
Most of the names that I can come up with are not substantially shorter or make the meaning any clearer than simply scanning the formula.  I think that, at least for the moment, I would leave this as is.  Would you agree?

Reply:
Introduced `denoms_clearable`!